bin/deploy

```diff
@@ -13,28 +13,77 @@ regions=(
   ap-southeast-2
 )
 
+
+function join_by()
+{
+  local IFS="$1"
+	shift
+	echo "$*"
+}
+
+function usage()
+{
+  regions_joined=$(join_by , ${regions[@]})
+  echo -e "Copies the CloudFormation templates into S3 buckets"
+  echo -e ""
+  echo -e "$0"
+  echo -e "\t-h --help"
+  echo -e "\t--bucket=$bucket"
+  echo -e "\t--regions=$regions_joined"
+  echo -e ""
+}
+
+while [ "$1" != "" ]
+do
+  PARAM=$(echo $1 | awk -F= '{print $1}')
+  VALUE=$(echo $1 | awk -F= '{print $2}')
+  case $PARAM in
+    -h | --help)
+      usage
+      exit
+      ;;
+    --bucket)
+      bucket=$VALUE
+      ;;
+    --regions)
+			unset regions
+			IFS=',' read -ra regions <<< "$VALUE"
+      ;;
+    *)
+      echo "ERROR: unknown parameter \"$PARAM\""
+      usage
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+
+set -o xtrace
```

Comment:
So, although having `xtrace` option on during args processing was too noisy, I could see some value in having it on during AWS CLI commands. Could be useful for the person running the command to understand what is going on.

That's why I moved it to L62.